packages/hardhat/test/integration/workflows/proxy_payment_addresses.js

```diff
@@ -0,0 +1,229 @@
+      contracts,
+      randomSignerFn,
+      local: { expectedProjectId },
+    }) =>
+      checkFn({
+        caller: randomSignerFn(),
+        contract: contracts.terminalDirectory,
+        fn: "terminalOf",
+        args: [expectedProjectId],
+        expect: contracts.terminalV1.address,
+      }),
+  },
+  {
+    description: "Deploy a proxy payment address",
+    fn: ({
+      executeFn,
+      deployer,
+      contracts,
+      randomStringFn,
+      local: { expectedProjectId },
+    }) =>
+      executeFn({
+        caller: deployer,
+        contract: contracts.proxyPaymentAddressManager,
+        fn: "deploy",
+        args: [expectedProjectId, randomStringFn()],
+      }),
+  },
+  {
+    description: "Make a payment to the proxy payment address",
+    fn: async ({
+      contracts,
+      BigNumber,
+      randomSignerFn,
+      local: { expectedProjectId },
+    }) => {
+      const [proxyPaymentAddress] =
+        await contracts.proxyPaymentAddressManager.addressesOf(
+          expectedProjectId
+        );
+      // An account that will be used to make payments.
+      const payer = randomSignerFn();
+
+      // Three payments will be made. Cant pay entire balance because some is needed for gas.
+      // So, arbitrarily divide the balance so that all payments can be made successfully.
+      const paymentValue = BigNumber.from(1);
+
+      await payer.sendTransaction({
+        to: proxyPaymentAddress,
+        value: paymentValue,
+      });
+
+      return { payer, paymentValue, proxyPaymentAddress };
+    },
+  },
+  {
+    description: "There should now be a balance in the proxy payment address",
+    fn: ({ verifyBalanceFn, local: { paymentValue, proxyPaymentAddress } }) =>
+      verifyBalanceFn({
+        address: proxyPaymentAddress,
+        expect: paymentValue,
+      }),
+  },
+  {
+    description: "Tap the proxy payment address",
+    fn: async ({ bindContractFn, local: { proxyPaymentAddress, payer } }) => {
+      // An account that will be used to make payments.
+      const proxyPaymentAddressContract = await bindContractFn({
+        contractName: "ProxyPaymentAddress",
+        address: proxyPaymentAddress,
+        signerOrProvider: payer,
+      });
+
+      await proxyPaymentAddressContract.tap();
+
+      return { payer, proxyPaymentAddress, proxyPaymentAddressContract };
+    },
+  },
+  {
+    description: "The balance should be empty in the proxy payment address",
+    fn: ({ verifyBalanceFn, local: { proxyPaymentAddress } }) =>
+      verifyBalanceFn({
+        address: proxyPaymentAddress,
+        expect: BigNumber.from(0),
+      }),
+  },
+  {
+    description:
+      "Make sure the correct number of tickets were printed for the proxy payment address",
+    fn: ({
+      checkFn,
+      contracts,
+      randomSignerFn,
+      local: { proxyPaymentAddress, expectedProjectId, paymentValue },
+    }) => {
+      const expectedNumTickets = paymentValue.mul(BigNumber.from(1000000));
```

Comment:
pass in `constants` and change BigNumber.from(1000000) to constants.InitialWeightMultiplier